ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java

```diff
@@ -84,34 +106,56 @@ private static String propertyName(final String name) {
           Importance.MEDIUM,
           LISTEN_PORT_DOC)
       .define(
-          KEY_PATH,
+          TLS_ENABLED,
+          Type.BOOLEAN,
+          DEFAULT_TLS_ENABLED,
+          Importance.MEDIUM,
+          TLS_ENABLED_DOC)
+      .define(
+          TLS_KEY_STORE_PATH,
+          Type.STRING,
+          "",
+          Importance.MEDIUM,
+          TLS_KEY_STORE_PATH_DOC)
+      .define(
+          TLS_KEY_STORE_PASSWORD,
           Type.STRING,
-          null,
+          "",
           Importance.MEDIUM,
-          KEY_PATH_DOC)
+          TLS_KEY_STORE_PASSWORD_DOC)
       .define(
-          CERT_PATH,
+          TLS_TRUST_STORE_PATH,
           Type.STRING,
-          null,
+          "",
           Importance.MEDIUM,
-          CERT_PATH_DOC)
+          TLS_TRUST_STORE_PATH_DOC)
+      .define(
+          TLS_TRUST_STORE_PASSWORD,
+          Type.STRING,
+          "",
+          Importance.MEDIUM,
+          TLS_TRUST_STORE_PASSWORD_DOC)
+      .define(
+          TLS_CLIENT_AUTH_REQUIRED,
+          Type.BOOLEAN,
+          DEFAULT_TLS_CLIENT_AUTH_REQUIRED,
+          Importance.MEDIUM,
+          TLS_CLIENT_AUTH_REQUIRED_DOC)
       .define(
           WORKER_POOL_SIZE,
           Type.INT,
           DEFAULT_WORKER_POOL_SIZE,
           Importance.MEDIUM,
-          WORKER_POOL_DOC);
+          WORKER_POOL_DOC)
+      .define(
+          MAX_PUSH_QUERIES,
+          Type.INT,
+          DEFAULT_MAX_PUSH_QUERIES,
+          Importance.MEDIUM,
+          MAX_PUSH_QUERIES_DOC);
```

Comment:
Add validator to ensure its not set to zero / negative.

Same for `WORKER_POOL_SIZE`

Reply:
Do you have an example of using a validator? I couldn't see one in KsqlConfig.

Reply:
nvm, found an example, and added some.